### ISSUE TITLE
Brainhack projects database

### ISSUE THREAD

**Issue by Remi-Gau** [Brainhack-Marseille/brainhack-marseille.github.io#28]
### Title

Brainhack projects database

### Leaders

Rémi Gau 
- https://allmylinks.com/remi-gau
- brainhack mattermost: remi-gau

### Collaborators

None at the moment

### Brainhack Global 2022 Event

Brainhack Marseille

### Project Description

**This is a remote project**

##### Problem

The brainhack has been running for more than 10 years, yet we do not have single centralised resource to show in a fairly exhaustive manner the diversity of projects that have happened over the past decade.

Having a quick way to create reports about the success of brainhack could make significantly easier for event organizer to look for funding.

##### Solution

In the past few years more and more events have started listing their projects as github issues.

This now makes it easier to 

- start creating a mini database of 
  - brainhack events, 
  - brainhack sites
  - brainhack projects

- to create an interactive dashboard to query and create visualizations of that database

## What makes your project special and exciting?

This is almost a blank slate project, so if you are afraid to break things: good news, there is (almost) nothing to break. 

If you know nothing about databases and dashboards: good news, me neither! I am using this project to learn about them.

### How to get started and Where to find key resources?

We will be working on this repo.

https://github.com/brainhackorg/brainhack_jupyter_book

Some of the data in this folder as TSV and JSON

https://github.com/brainhackorg/brainhack_jupyter_book/tree/main/data

There are a couple of scripts to help list project issues from a repo and create a table of all projects

https://github.com/brainhackorg/brainhack_jupyter_book/blob/main/scripts

/get_projects_issues.pyhttps://github.com/brainhackorg/brainhack_jupyter_book/blob/main/scripts/create_projects_table.py

### Link to project repository/sources

https://github.com/brainhackorg/brainhack_jupyter_book

### Goals for Brainhack Global

- for past brainhack events
  - create a github repo if they do not have one
  - use an issue template to create one github issue for each project of those events (if they don't have one)
  - use automatic issue labelling to label projects topic, programming language, ...

- collect all projects infos for each event using the github API

- create a table of all projects

- create an interactive dashboard

### Good first issues

good first issues are listed here

https://github.com/brainhackorg/brainhack_jupyter_book/labels/good%20first%20issue

### Communication channels

We will be using the mattermost channel that is used by the "brainhack paper jupyter book".

https://mattermost.brainhack.org/brainhack/channels/brainhack_paper_jupyterbook

### Skills

- you know some python
- you can open a github issue
- you know about databases / APIs / interactive dashboards or want to learn about them

### Onboarding documentation

None at the moment

### What will participants learn?

- use github API to collect informations about repositories
- making a tiny cute database
- management of github repos and some automation related to it
- create vizualisation dashboard

### Data to use

Data is currently stored on this repo as TSV and JSON

https://github.com/brainhackorg/brainhack_jupyter_book/tree/main/data

Part of this project is to make this data grow.

### Number of collaborators

2

### Credit to collaborators

Be added to the brainhack jupyter book contributors list

### Image

Leave this text if you don't have an image yet.

### Type

other

### Development status

0_concept_no_content

### Topic

data_visualisation

### Tools

other

### Programming language

Python, other

### Modalities

not_applicable

### Git skills

0_no_git_skills, 1_commit_push, 3_continuous_integration

### Anything else?

Not for now

### Things to do after the project is submitted and ready to review.

- [ ] Add a comment below the main post of your issue saying: `Hi @brainhackorg/project-monitors my project is ready!`
- [ ] Twitter-sized summary of your project pitch.

**Comment by StanSStanman** [comment]
Thank you @Remi-Gau for submitting your project to BrainHack Marseille 2022! :tada::tada::tada:
Your project have been added to our website.

Since no image was provided, we added a default one. Feel free to send us a new image, and we will update it. 

Given the nature of the project, I have a question from my side: can tagging the brainhack global organization in the local issues submission be a solution for adding them on the global website? Or we should ask participants to duplicate the issue on the brainhack global git?

Ruggero for the BHM team.
